File: tests/test_code/php/rce_tests.php
<?php
// VULNERABLE: INSECURE_DESERIALIZATION
$data = $_GET['data'];
unserialize($data);

// VULNERABLE: SSTI (Twig example)
$loader = new \Twig\Loader\ArrayLoader();
$twig = new \Twig\Environment($loader);
$template = $twig->createTemplate("Hello " . $_GET['name']);
echo $template->render();
?>

File: tests/test_code/python/rce_tests.py
import pickle
import yaml
import jinja2
from flask import Flask, request

app = Flask(__name__)

@app.route('/unpickle')
def unpickle():
    data = request.args.get('data')
    # VULNERABLE: INSECURE_DESERIALIZATION
    pickle.loads(data)
    return "Done"

@app.route('/yaml')
def load_yaml():
    data = request.args.get('config')
    # VULNERABLE: INSECURE_DESERIALIZATION (if not using SafeLoader)
    yaml.load(data)
    return "Loaded"

@app.route('/template')
def template():
    user_input = request.args.get('name')
    # VULNERABLE: SSTI
    template = jinja2.Environment().from_string("Hello " + user_input)
    return template.render()
